akka-persistence-typed/src/main/scala/akka/persistence/typed/javadsl/ReplicatedEventSourcing.scala

```diff
@@ -118,7 +121,20 @@ object ReplicatedEventSourcing {
       allReplicasAndQueryPlugins: JMap[ReplicaId, String],
       eventSourcedBehaviorFactory: JFunction[ReplicationContext, EventSourcedBehavior[Command, Event, State]])
       : EventSourcedBehavior[Command, Event, State] = {
-    val context = new ReplicationContextImpl(replicationId, allReplicasAndQueryPlugins.asScala.toMap)
+    val plugins = allReplicasAndQueryPlugins.asScala.toMap
+    val context =
+      new ReplicationContextImpl(replicationId, plugins.keySet, system => new JournalEventHub(plugins)(system))
+    eventSourcedBehaviorFactory(context)
+  }
+
+  @ApiMayChange
+  def withCustomReplication[Command, Event, State](
```

Comment:
Or we make it a configuration property? I think an ActorSystem global setting would be enough.

Reply:
let's try it out when we see the full picture, might not work with config, you know we had to pass in the ReadJournal instance to the source provider

Reply:
Yeah, that's what I was thinking, we'll need a ton of setup for the thing we feed in, so I think config here would be messy, but maybe config "there" and then pass the instance you created from config.